src/javascript/binary/websocket_pages/user/account/profit_table/profit_table.init.js

```diff
@@ -106,33 +100,32 @@ const ProfitTableWS = (function() {
         });
     };
 
-    const initSocket = function() {
-        BinarySocket.init({
-            onmessage: function(msg) {
-                const response = JSON.parse(msg.data);
-
-                if (response) {
-                    const type = response.msg_type;
-                    if (type === 'profit_table') {
-                        ProfitTableWS.profitTableHandler(response);
-                        showLocalTimeOnHover('td.buy-date,td.sell-date');
-                    } else if (type === 'oauth_apps') {
-                        addTooltip(ProfitTableUI.setOauthApps(buildOauthApps(response.oauth_apps)));
-                    }
-                }
-            },
+    const getProfitTable = (opts) => {
+        const req = { profit_table: 1, description: 1 };
+
+        if (opts) $.extend(true, req, opts);
+
+        BinarySocket.send(req).then((response) => {
+            ProfitTable.profitTableHandler(response);
+            showLocalTimeOnHover('td.buy-date,td.sell-date');
+            $('.barspinner').addClass('hidden');
+        });
+    };
+
+    const initSocket = () => {
```

Comment:
can you change the name of this function? it's not initializing socket anymore, something like get apps or similar would be more appropriate